resources/home/dnanexus/run_workflows/utils/utils.py

```
@@ -485,96 +166,531 @@
+        else:
+            raise AssertionError(
+                (
+                    f"'{pattern}' doesn't match the expected pattern of a "
+                    "sample name or an assay code"
+                )
+            )
+
+    return list(new_patterns)
+
+
+def exclude_samples(samples, patterns=[]):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)